.changeset/clever-emus-roll.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Add support for setting timeout in client:idle directive
```

Comment:
``````suggestion
Adds a new, optional property `timeout` for the `client:idle` directive. 

This value allows you to specify a maximum value to wait, in milliseconds, before hydrating a UI framework component, even if the page is not yet done with its initial load. This means you can delay hydration for lower-priority UI elements with more control to ensure your element is interactive within a specified time frame.

```astro
<ShowHideButton client:idle={{timeout: 500}} />
```
``````
You'll need to check/correct my facts/nuance here, but I'm thinking something like this is a nice way to present this handy new feature! Especially for a minor release, we do like to hype things up a bit more. So what about something in this style?